hedyorg/hedy
# 💻 variable list is shown after change to variable

the variable list is opened immediately, without the need to click the label button first, when a variable changes. This might be annoying with a lot of variable changes though, since after clicking this view to close, it opens back up on itself if a variable changes, but it was suggested in issue #5092 that extra attention would be good. The question is, is this too much?

Start the name of the PR with one of the relevant prefixes in the title of the PR:

* 🖊️ -> changes related to grammars or the transpiler
* 🪲 -> solving a bug
* 🚚 -> changes unrelated to shipping, such as mergify scripts
* 🧹 -> refactorings (changes that do not change functionality and are only a cleanup of code)
* 💻 -> improvements and changes of the Hedy website
* 🧪 -> changes made to the test suite
* 📜 -> changes related to content

Fixes #5147  
Fixes #5092 

**How to test**

Follow these steps to verify this PR works as intended:

* Go to level 5 of Hedy (or another level in which variables exist)
* Create a program with at least two variables, for example
```
naam is Hedy
leeftijd is 15
leeftijd is 40
print naam is leeftijd jaar oud
```
* Run this program step-by-step
* Check if at first, when there are no variables yet, you do not see the blue label for the variable list and also not the list itself, in the top right corner of the output view, but when there is a variable, the button appears and the variable list is shown
* Check that if you click the label button, the list disappears, but the button stays. If you click it again, the list comes back
* Check that if the button is clicked and the variable list is hidden, the list opens automatically back up if a variable changes.

**Checklist**
Done? Check if you have it all in place using this list: (mark with x if done)

- [ x] Contains one of the PR categories in the name
- [ x] Describes changes in the format above
- [ x] Links to an existing issue or discussion
- [ x] Has a "How to test" section

If you're unsure about any of these, don't hesitate to ask. We're here to help!
